game-core/src/main/java/games/strategy/triplea/attachments/TerritoryAttachment.java

```diff
@@ -78,19 +80,15 @@ public static boolean doWeHaveEnoughCapitalsToProduce(
    * we find that we don't own. If a capital has no neighbor connections, it will be sent last.
    */
   public static @Nullable Territory getFirstOwnedCapitalOrFirstUnownedCapital(
-      final GamePlayer player, final GameData data) {
+      final GamePlayer player, final GameMap gameMap) {
     final List<Territory> capitals = new ArrayList<>();
     final List<Territory> noNeighborCapitals = new ArrayList<>();
-    for (final Territory current : data.getMap().getTerritories()) {
+    for (final Territory current : gameMap.getTerritories()) {
       final TerritoryAttachment ta = TerritoryAttachment.get(current);
       if (ta != null && ta.getCapital() != null) {
-        final GamePlayer whoseCapital = data.getPlayerList().getPlayerId(ta.getCapital());
-        if (whoseCapital == null) {
```

Comment:
These checks are redundant since `setCapital` doesn't allow unknown player names.  So `ta.getCapital()` will either return a null or will return a String that is a valid player name.